### ISSUE TITLE
Investigate Repeated Actions

### ISSUE THREAD

**Issue by mbecker12** [mbecker12/surface-rl-decoder#86]
The agent still seems to select only one action in one episode, it even seems to happen across different episodes.

- [x] Log the index of the chosen action in the evaluation step.
- [x] Look at the difference between the max q value and the second largest 
- [x] Consider outputting the distribution of q values (need to think about how; across an episode, in one step; across all episodes?)
- [x] Schedule some training runs w/ high verbosity to look at network param distribution, to see if it learns
- [x] Can a higher learning rate help?
- [x] Lower target update steps again
- [x] Change discount factor

**Comment by mbecker12** [comment]
Consider implementing #29 as a handy way to keep track of the action history. One could then append the grid-action-history to the current state.
Maybe one would need to account for the appended layer (i.e. the grid action history) in the nn architecture. We probably want to treat the grid action history a little differently and feed it through another network branch before merging it to the final layers

**Comment by mbecker12** [comment]
Job 59889 on Alvis revealed that at least in the beginning, the first layer parameters don't seem to change.
Could be a cause of vanishing gradients

**Comment by mbecker12** [comment]
Note, it might be the case that the 'first layer' under investigation in tensorboard, refers to the first layer that is initialized in the agent's init function. As of now, this would be a layer acting on a split state. The split functionality has been disabled for the most recent training runs in order to make the models simpler. Therefore, this layer might intentionally not be trained.
Need to check the order of layers in the network, and log the evolution of the 'real' active first layer

**Comment by mbecker12** [comment]
Increasing both the state values and the intermediate reward for annihilating a syndrome seemed to have helped, at least as a short term solution. See #87
